backend/composer/services/cs_ingestion/cs_ingestion_services.py

```diff
@@ -40,253 +39,351 @@
-        invalid_entities.append({"entity": statement[ORIGIN][0], "prop": ORIGIN})
-    if has_prop(statement[DESTINATION]) and not_found_entity(statement[DESTINATION][0][ENTITY_URI]):
-        invalid_entities.append({"entity": statement[DESTINATION][0][ENTITY_URI], "prop": DESTINATION})
-    if has_prop(statement[VIAS]):
-        for entity in statement[VIAS]:
-            if not_found_entity(entity[ENTITY_URI]):
-                invalid_entities.append({"entity": entity[ENTITY_URI], "prop": VIAS})
-
-    if len(invalid_entities) > 0:
-        logging.warning(
-            f'Skip statement {statement[LABEL]} due to the following entities not found in composer db: {invalid_entities}')
-        return True
-    return False
-
-
-def has_multiple_destination(statement):
-    if len(statement[DESTINATION]) > 1:
-        logging.warning(f'Skip statement {statement[LABEL]} due to multiple destinations')
-        return True
-    else:
-        return False
-
-
-def has_invalid_sex(statement):
-    if has_prop(statement[SEX]) and len(Sex.objects.filter(ontology_uri=statement[SEX][0])) == 0:
-        logging.warning(f'Skip statement {statement[LABEL]} due to sex {statement[SEX][0]} not found in composer db')
-        return True
-    else:
-        return False
-
-
-def has_invalid_species(statement):
-    if has_prop(statement[SPECIES]):
-        for s in statement[SPECIES]:
-            try:
-                Specie.objects.get(ontology_uri=s)
-                return False
-            except:
-                logging.warning(f'Skip statement {statement[LABEL]} due to specie {s} not found in composer db')
-                return True
-    return False
+def ingest_statements():
+    statements_list = get_statements_from_neurondm()
 
+    valid_statements = validate_statements(statements_list)
+    for statement in valid_statements:
+        sentence, sentence_created = get_or_create_sentence(statement)
 
-def get_valid_phenotypes(statement):
-    valid_phenotypes = []
-    if has_prop(statement[PHENOTYPE]):
-        for p in statement[PHENOTYPE]:
-            try:
-                phenotype = Phenotype.objects.get(ontology_uri=p)
-                valid_phenotypes.append(phenotype)
-            except:
-                logging.warning(f'Skip phenotype {p} at statement {statement[LABEL]} not found in composer db ')
-    return valid_phenotypes
+        connectivity_statement, statement_created = get_or_create_connectivity_statement(statement, sentence)
 
+        if not statement_created:
+            if should_overwrite(connectivity_statement, statement):
+                defaults = generate_connectivity_statement_defaults(statement, sentence)
+                ConnectivityStatement.objects.filter(id=connectivity_statement.id).update(**defaults)
+                update_many_to_many_fields(connectivity_statement, statement)
+                add_ingestion_system_note(connectivity_statement)
```

Comment:
This adds a new note to inform the users about an overwrite that happened due to the ingestion in progress

Reply:
@afonsobspinto I'm not a big fan of changing the records via the `.update` method
please have a look here: https://docs.djangoproject.com/en/5.0/topics/db/queries/#updating-multiple-objects-at-once
fourth alinea the `Beware part`

better use the `.update_or_create(filter,filter,defaults=defaults`

Reply:
the latter will trigger the save methods and signals (that are connected to the model)

Reply:
Thank you django specialist @zsinnema 😅